ios/src/TGMapViewController.mm

```diff
@@ -546,10 +549,11 @@ - (void)respondToShoveGesture:(UIPanGestureRecognizer *)shoveRecognizer
     // don't trigger shove on single touch gesture
     if ([shoveRecognizer numberOfTouches] == 2) {
         if (self.gestureDelegate && [self.gestureDelegate respondsToSelector:@selector(recognizer:didRecognizeShoveGesture:)]) {
-            [self.gestureDelegate mapView:self recognizer:shoveRecognizer didRecognizeShoveGesture:displacement];
-        } else {
-            self.map->handleShoveGesture(displacement.y);
+            if (![self.gestureDelegate mapView:self recognizer:shoveRecognizer didRecognizeShoveGesture:displacement]) {
```

Comment:
Ah I am following now. Part of the issue I'm having here is that a "shove" gesture is not a gesture built into iOS, so users aren't necessarily going to know how to react to what a shove gesture actually is. 

For that kind of a pattern, Apple normally separates it into separate delegate function calls. A good example is `UITableViewDelegate's` different `should*` functions versus `did*` functions - https://developer.apple.com/reference/uikit/uitableviewdelegate

So maybe that's something we should do here as well? We can have a "shouldRecognizeShoveGesture" and a "didRecognizeShoveGesture". The `shouldRecognize` functions would pass along things like "numberOfTouches` and other important parameters, and expect a boolean return. If the user returns `true` or doesn't implement the `shouldRecognize` then Tangram will handle the recognition, and then call the "didRecognize".

I think this will also clear up @blair1618's question below.